include/ginkgo/core/matrix/csr.hpp

```diff
@@ -207,7 +222,11 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
                 for (size_type i = 1; i < nwarps; i++) {
                     srow[i] += srow[i - 1];
                 }
-                *mtx_srow = srow_host;
+                if (!is_srow_on_host) {
+                    *mtx_srow = srow_host;
+                }
+            } else {
+                GKO_NOT_SUPPORTED(nwarps);
```

Comment:
If without this, would it have any problem?
I would move it into cuda kernel.
In benchmark/spmv without compiling cuda module, it would create matrix on the reference with `nwarps == 0` such that throw `GKO_NOT_SUPPORT`

Reply:
You are correct, moving it there is the right place. Especially empty matrices might be a problem with this handling, so I moved it to the cuda module, as you suggested.